dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -1,70 +1,175 @@
-        return self.parsed_model["config"].get("job_cluster_config", {})
+    def submit(self, compiled_code: str) -> None:
+        self.command_submitter.submit(compiled_code)
 
-    def __init__(self, parsed_model: Dict, credentials: DatabricksCredentials) -> None:
-        self.credentials = credentials
-        self.identifier = parsed_model["alias"]
-        self.schema = parsed_model["schema"]
-        self.database = parsed_model.get("database")
-        self.parsed_model = parsed_model
-        use_user_folder = parsed_model["config"].get("user_folder_for_python", False)
 
-        self.check_credentials()
+class PythonCommandSubmitter(PythonSubmitter):
+    """Submitter for Python models using the Command API."""
 
-        self.api_client = DatabricksApiClient.create(
-            credentials, self.get_timeout(), use_user_folder
+    def __init__(
+        self, api_client: DatabricksApiClient, tracker: PythonRunTracker, cluster_id: str
+    ) -> None:
+        self.api_client = api_client
+        self.tracker = tracker
+        self.cluster_id = cluster_id
+
+    @override
+    def submit(self, compiled_code: str) -> None:
+        context_id = self.api_client.command_contexts.create(self.cluster_id)
+        command_exec: Optional[CommandExecution] = None
+        try:
+            command_exec = self.api_client.commands.execute(
+                self.cluster_id, context_id, compiled_code
+            )
+
+            self.tracker.insert_command(command_exec)
+            # poll until job finish
+            self.api_client.commands.poll_for_completion(command_exec)
+
+        finally:
+            if command_exec:
+                self.tracker.remove_command(command_exec)
+            self.api_client.command_contexts.destroy(self.cluster_id, context_id)
+
+
+class PythonNotebookUploader:
+    """Uploads a compiled Python model as a notebook to the Databricks workspace."""
+
+    def __init__(
+        self, api_client: DatabricksApiClient, catalog: str, schema: str, identifier: str
+    ) -> None:
+        self.api_client = api_client
+        self.catalog = catalog
+        self.schema = schema
+        self.identifier = identifier
+
+    @staticmethod
+    def create(
+        api_client: DatabricksApiClient, parsed_model: ParsedPythonModel
+    ) -> "PythonNotebookUploader":
+        return PythonNotebookUploader(
+            api_client,
+            parsed_model.catalog,
+            parsed_model.schema_,
+            parsed_model.identifier,
         )
 
-        self.job_grants: Dict[str, List[Dict[str, Any]]] = self.workflow_spec.pop("grants", {})
+    def upload(self, compiled_code: str) -> str:
+        """Upload the compiled code to the Databricks workspace."""
+        workdir = self.api_client.workspace.create_python_model_dir(self.catalog, self.schema)
+        file_path = f"{workdir}{self.identifier}"
+        self.api_client.workspace.upload_notebook(file_path, compiled_code)
```

Comment:
do you need to rollback (delete pytho_model_dir) when `upload_notebook` fails?

Reply:
I don't think we want to delete the directory because there could be other python model notebooks in the same directory